src/keyboard_x.cc

```diff
@@ -352,8 +354,6 @@ void* listenToXEvents(void *arg) {
   KbState lastState;
   readKbState(display, &lastState);
 
-  NAPI_CALL(data->env, napi_acquire_threadsafe_function(data->tsfn));
-
   XkbEvent event;
   KbState currentState;
   while (true) {
```

Comment:
Since we are using this as a long running event watcher thread, we cannot break this due to the nature of Xlib calls. I added the thread cleanup hooks when `tsfn` gets destroyed by node environment teardown.

Reply:
:+1: I also read about `pthreads` that this child thread will exit as soon as `exit()` is called, so I hope that this does not result in any leaks. We are calling this listener only once, from the main process.